samples/subsys/ipc/ipc_service/static_vrings/remote/src/main.c

```diff
@@ -233,14 +244,32 @@ static void ipc1_entry(void *dummy0, void *dummy1, void *dummy2)
 	k_sem_take(&ipc1_bound_sem, K_FOREVER);
 
 	while (message < 99) {
+		uint32_t len = 0;
+		void *data;
+
 		k_sem_take(&ipc1_data_sem, K_FOREVER);
-		message = ipc1_received_data;
+
+		message = *((uint8_t *) recv_data);
+
+		ret = ipc_service_release_rx_buffer(&ipc1_ept, recv_data);
+		if (ret < 0) {
+			printk("ipc_service_release_rx_buffer failed with ret %d\n", ret);
+			break;
+		}
 
 		printk("REMOTE [1]: %d\n", message);
 
 		message++;
 
-		ret = ipc_service_send(&ipc1_ept, &message, sizeof(message));
+		ret = ipc_service_get_tx_buffer(&ipc1_ept, &data, &len, K_FOREVER);
+		if (ret < 0) {
+			printk("ipc_service_get_tx_buffer failed with ret %d\n", ret);
+			break;
+		}
+
+		*((unsigned char *) data) = message;
```

Comment:
If `len = 0`, you should verify `len` value after calling `ipc_service_get_tx_buffer()` and before filling the buffer.